build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt

```diff
@@ -317,6 +380,55 @@ class PerformanceTestExtension(
         registerTestProject(testProject, T::class.java, configuration)
 
     fun <T : Task> registerTestProject(testProject: String, type: Class<T>, configurationAction: Action<in T>): TaskProvider<T> {
+        return doRegisterTestProject(testProject, type, configurationAction)
+    }
+
+    fun <T : Task> registerAndroidTestProject(testProject: String, type: Class<T>, configurationAction: Action<in T>): TaskProvider<T> {
+        return doRegisterTestProject(testProject, type, configurationAction) {
+            // AndroidStudio jvmArgs could be set per project, but at the moment that is not necessary
+            jvmArgumentProviders.add(getAndroidProjectJvmArguments(defaultAndroidStudioJvmArgs))
+            environment("JAVA_HOME", javaLauncher.get().metadata.installationPath.asFile.absolutePath)
+        }
+    }
+
+    private
+    fun getAndroidProjectJvmArguments(androidStudioJvmArgs: List<String>): CommandLineArgumentProvider {
+        val unpackAndroidStudio = project.tasks.named("unpackAndroidStudio", Copy::class.java)
+        val androidStudioInstallation = project.objects.newInstance<AndroidStudioInstallation>().apply {
+            studioInstallLocation.fileProvider(unpackAndroidStudio.map { it.destinationDir })
+        }
+        val autoDownloadAndroidStudio = isCiServer
+        val runInHeadlessMode = isCiServer
+        return object : AndroidStudioSystemProperties(androidStudioInstallation, autoDownloadAndroidStudio, project.providers) {
+            override fun asArguments(): Iterable<String> {
+                val systemProperties = mutableListOf<String>()
+                if (autoDownloadAndroidStudio) {
+                    val androidStudioPath = androidStudioInstallation.studioInstallLocation.asFile.get().absolutePath
+                    val macOsAndroidStudioPath = "$androidStudioPath/Android Studio.app"
+                    val macOsAndroidStudioPathPreview = "$androidStudioPath/Android Studio Preview.app"
+                    val windowsAndLinuxPath = "$androidStudioPath/android-studio"
+                    val studioHome = when {
+                        isMacOsX && File(macOsAndroidStudioPath).exists() -> macOsAndroidStudioPath
+                        isMacOsX -> macOsAndroidStudioPathPreview
+                        else -> windowsAndLinuxPath
+                    }
+                    systemProperties.add("-Dstudio.home=$studioHome")
+                } else {
+                    System.getProperty("studio.home")?.let {
```

Comment:
Is `studio.home` now a build parameter as well and should be treated as such? I.e. should it be a constant in `BuildParams`?

Reply:
Moved to `BuildParams`